src/Traits/HasNameTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace WikibaseSolutions\CypherDSL\Traits;
+
+use LogicException;
+use function bin2hex;
+use function ceil;
+use function openssl_random_pseudo_bytes;
+use function substr;
+
+trait HasNameTrait
+{
+    use ErrorTrait;
+
+    private string $name;
+
+    /**
+     * Returns the length in bytes of the auto-generated part of a name.
+     *
+     * @return int
+     */
+    public static function automaticVariableLength(): int
+    {
+        return 32;
+    }
+
+    /**
+     * Returns the name.
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        if (!isset($this->name)) {
+            throw new LogicException('Name is not yet configured. Please call `configureName` to setup a name.');
```

Comment:
You probably just want to call configureName here instead of giving an exception